spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.stream.binder.pubsub;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubConsumerProperties;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubProducerProperties;
+import org.springframework.cloud.stream.binder.AbstractBinderTests;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.Spy;
+
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+/**
+ * Integration tests that require the Pub/Sub emulator to be installed and only run if the
+ * GCP_PUBSUB_EMULATOR envvar is present.
+ *
+ * @author João André Martins
+ */
+public class PubSubMessageChannelBinderTests extends AbstractBinderTests<PubSubTestBinder,
+		ExtendedConsumerProperties<PubSubConsumerProperties>,
+		ExtendedProducerProperties<PubSubProducerProperties>> {
+
+	@Rule
+	public PubSubEmulator emulator = new PubSubEmulator();
+
+	private PubSubTestBinder binder;
+
+	public PubSubMessageChannelBinderTests() {
+		this.binder = new PubSubTestBinder(this.emulator.getPort());
+	}
+
+	@BeforeClass
+	public static void enableTests() {
+		assumeThat(System.getenv("GCP_PUBSUB_EMULATOR")).isNotNull();
```

Comment:
Why do I need to set the ENV var if I start emulator manually from the rule?
Can't we just rely on the fact that emulator started or not?
For example:
```
logger.warn("Tests not running because no broker on " + url + ":", e);
assumeNoException(e);
```
in that `PubSubEmulator` instead.

Reply:
This is what triggers the emulator running or not.
If `GCP_PUBSUB_EMULATOR` isn't an envvar, the rule isn't triggered.

Reply:
That's wrong by design again. 
What happens if I have `GCP_PUBSUB_EMULATOR` env var, but don't have emulator?

More over it would be much better to be as friendly for external contributors as possible.
IMO just a single rule state is enough to determine if we need to run the tests or not.

I don't have any experience why I would prefer do not run tests at all during project build (unless it is just about docs building). So, if I have emulator, I definitely would like to run tests in the project.

Like I have all the Redis, MongoDB, Kafka, RabbitMQ etc. in my machine and I'm pretty happy to have all the test passes. If I don't have some resource (e.g. Oracle DB), the single rule makes a decision do not run tests. That's all.

I don't get it why do we need one more level of the management...

Reply:
According to https://cloud.google.com/pubsub/docs/emulator#pubsub-emulator-java, the Pub/Sub emulator is started if `PUBSUB_EMULATOR_HOST` is populated.
So I'll change this to the new envvar and delete `PubSubEmulator`. How does that sound?

Reply:
Sounds good, but wait. Aren't you going to run emulator from the test suit any more then?
Well, actually that's fine with me, too. At least it is similar to what I do with `kinesalite` for testing AWS Kinesis. However I think I could start it from the `/npm` the same way you do here in the rule 😄

Reply:
If the test starts the emulator, I think there might be some issues with the `@BeforeClass` not catching the populated envvar, because the emulator rule hasn't run yet.
However, I think this deserves some more thought, so I'll file an issue to consider the test running the emulator on its own.